libcontainer/cgroups/utils.go

```diff
@@ -494,6 +494,21 @@ func getHugePageSizeFromFilenames(fileNames []string) ([]string, error) {
 	return pageSizes, nil
 }
 
+// WalkCgroups calls walkFn on the cgroup directory at path and on all its
+// subcgroups.
+func WalkCgroups(path string, walkFn func(path string) error) error {
+	return filepath.Walk(path, func(p string, info os.FileInfo, iErr error) error {
+		dir, file := filepath.Split(p)
+		if file != CgroupProcesses {
+			return nil
+		}
+		if iErr != nil {
```

Comment:
Error handling (if you really need it, I'm not sure) should be done before filename comparison.